### PR TITLE
docs: update readme table of contents

### LINKED ISSUE #483: Readme table of contents out-of-date with readme content

### PR DESCRIPTION
**Pull request by mightyiam**
Fixes #483.

Made with https://marketplace.visualstudio.com/items\?itemName\=yzhang.markdown-all-in-one .

### CODE REVIEW

**Comment by paldepind** [comment]
Looks good!

A few thoughts:

* Isn't it weird to include an entry for table of contents in the table of contents?
* What about only including top-level headers `##` in the ToC? Currently, it becomes quite long with all the subsections.

File: README.md [hunk 1]
```diff
@@ -12,17 +12,53 @@ and performance.
...
-* [Introduction](#introduction)
-* [Features](#features)
-* [Inline example](#inline-example)
-* [Examples](#examples)
-* [Core documentation](#core-documentation)
-* [Modules documentation](#modules-documentation)
-* [Helpers](#helpers)
-* [Virtual Node documentation](#virtual-node)
-* [Structuring applications](#structuring-applications)
+<!-- Tip: initially generated with https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one -->
```
Comment (paldepind) [inline README.md]: What about adding something like a `**Table of Contents**` non-header here?

**Comment by paldepind** [comment]
Regarding the length, I think its a lot to scroll over. Especially for someone who don't know what Snabbdom is and who just wants to read the introduction.

I don't think the subitems adds much value. Consider for instance this section for Virtual Node.

Screenshot 2019-11-26 at 17 55 43

Does it really help to see all the vnode properties in the ToC?

Having a "Table of Contents" header above the table of contents is nice IMO. I like the idea of generating the ToC automatically, but, I don't think we should compromise on the quality of the readme due limits in the tool.

**Comment by mightyiam** [comment]
Thank you, @paldepind. Closing in favor of #490.

**Pull request closed without merging**